### PR TITLE
 Enable adding annotation tag filters via clicking

### LINKED ISSUE #4131: Add tags to the annotation hover thingy
as a chronograf user
when i hover over an annotation and view the name
then i would like to also see the tags associated with that annotation

### PR DESCRIPTION
**Pull request by chnn**
Closes #4131

filter by clicking

cc @alexpaxton

### CODE REVIEW

File: ui/src/shared/components/AnnotationControlBar.tsx [hunk 1]
```diff
@@ -74,24 +75,22 @@ class AnnotationControlBar extends PureComponent<Props> {
...
+    await onUpdateTagFilterAsync(source.links.annotations, dashboardID, t)
   }
 
   private handleDeleteTagFilter = async (t: TagFilter): Promise<void> => {
-    const {dashboardID, onDeleteTagFilter, onRefreshAnnotations} = this.props
+    const {source, dashboardID, onDeleteTagFilterAsync} = this.props
 
-    onDeleteTagFilter(dashboardID, t)
-    await onRefreshAnnotations()
+    onDeleteTagFilterAsync(source.links.annotations, dashboardID, t)
```
Comment (jaredscheib) [inline ui/src/shared/components/AnnotationControlBar.tsx]: would an `await` be appropriate here?

**Comment by chnn** [comment]
@jaredscheib Ready for another review

File: ui/src/dashboards/selectors/index.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import {DEFAULT_TIME_RANGE} from 'src/shared/data/timeRanges'
+import idNormalizer, {TYPE_ID} from 'src/normalizers/id'
+import {State as DashTimeState} from 'src/dashboards/reducers/dashTimeV1'
+
+export const getTimeRange = (state: {dashTimeV1: DashTimeState}, dashboardID) =>
+  state.dashTimeV1.ranges.find(
```
[the remark this answers is not in the document]
Comment (jaredscheib) [reply ui/src/dashboards/selectors/index.tsx]: word

File: ui/src/shared/actions/annotations.ts [hunk 3]
```diff
@@ -250,14 +254,20 @@ export const addAnnotationAsync = (
...
 ) => async dispatch => {
   dispatch(addAnnotation(annotation))
-  const savedAnnotation = await api.createAnnotation(createUrl, annotation)
-  dispatch(deleteAnnotation(annotation))
-  dispatch(addAnnotation(savedAnnotation))
+
+  try {
+    const savedAnnotation = await api.createAnnotation(createUrl, annotation)
+
+    dispatch(deleteAnnotation(annotation))
```
Comment (jaredscheib) [inline ui/src/shared/actions/annotations.ts]: are these two dispatches batched such that the user wouldn't see it disappear momentarily? i don't think they are but not 100% positive. what do you think about a single dispatch of `updateAnnotation` that replaces it with a single action? that's the pattern we've used previously as well.

File: ui/src/shared/actions/annotations.ts [hunk 4]
```diff
@@ -283,15 +296,61 @@ export const getAnnotationsAsync = (
...
 }
 
 export const updateAnnotationAsync = (
   annotation: Annotation
 ) => async dispatch => {
-  await api.updateAnnotation(annotation)
-  dispatch(updateAnnotation(annotation))
+  try {
+    await api.updateAnnotation(annotation)
+    dispatch(updateAnnotation(annotation))
```
Comment (jaredscheib) [inline ui/src/shared/actions/annotations.ts]: you could dispatch this first for the all optimists out there

File: ui/src/shared/actions/annotations.ts [hunk 5]
```diff
@@ -283,15 +296,61 @@ export const getAnnotationsAsync = (
...
+  } catch {
+    dispatch(notify(annotationsError('Error saving annotation')))
```
Comment (jaredscheib) [inline ui/src/shared/actions/annotations.ts]: and if the update fails, should we update it back to its original state? previously this has been done by re-updating to whatever first came in to the function, and then the update on success would be an update with whatever is returned from the success. unlesss they're the same

**Pull request merged**